wagtail/models/__init__.py

```diff
@@ -901,30 +1068,26 @@ def save_revision(
         if clean:
             self.full_clean()
 
-        new_comments = getattr(self, COMMENTS_RELATION_NAME).filter(pk__isnull=True)
-        for comment in new_comments:
+        self._new_comments = getattr(self, COMMENTS_RELATION_NAME).filter(
+            pk__isnull=True
+        )
```

Comment:
Need to set it as an instance attribute as it's used in `_after_save_revision()`. Not sure if this is a good idea, though...